homeassistant/components/holiday/calendar.py

```diff
@@ -0,0 +1,106 @@
+"""Holiday Calendar."""
+from __future__ import annotations
+
+from datetime import datetime
+
+from holidays import country_holidays
+
+from homeassistant.components.calendar import CalendarEntity, CalendarEvent
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_COUNTRY
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.util import dt as dt_util
+
+from .const import CONF_PROVINCE, DOMAIN
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Holiday Calendar config entry."""
+    country: str = config_entry.data[CONF_COUNTRY]
+    province: str | None = config_entry.data.get(CONF_PROVINCE)
+
+    entity = HolidayCalendarEntity(
+        config_entry.title, country, province, unique_id=config_entry.entry_id
+    )
+    async_add_entities([entity], True)
+
+
+class HolidayCalendarEntity(CalendarEntity):
+    """Representation of a Holiday Calendar element."""
+
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        name: str,
+        country: str,
+        province: str | None,
+        unique_id: str,
+    ) -> None:
+        """Initialize HolidayCalendarEntity."""
+        self._country = country
+        self._province = province
+        self._event: CalendarEvent | None = None
+        self._attr_name = name
+        self._attr_unique_id = unique_id
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, self._attr_unique_id)},
+            name=self._attr_name,
+        )
+
+    @property
+    def event(self) -> CalendarEvent | None:
+        """Return the next upcoming event."""
+        obj_holidays = country_holidays(
+            self._country,
+            subdiv=self._province,
+            years=dt_util.now().year,
+            language=self.hass.config.language,
```

Comment:
I didn't look myself yet (as I will do it for workday) but is the languages really matching up and what happens if the country does not support the language set in HA?

Reply:
> If the language translation is not supported the original holiday names will be used.

(https://python-holidays.readthedocs.io/en/latest/api.html)

Reply:
I know in `holidays` that US english is `en_US` and just looking in HA it's `en` so this won't match up I guess?